Add DSL documentation feature (#1572)
theforeman/theforeman.org: _includes/manuals/nightly/1.2_release_notes.md

### Headline features

#### Templates DSL documentation

Foreman now has documentation for DSL used for templates writing. This should
make template writing process easier for non-Rubyists as well as for people who

> do know less about certain methods Foreman provides. Documentation is available

Reviewer: ```suggestion
aren't familiar with all the methods Foreman provides. Documentation is available locally
```

> under `https://<fqdn>/templates_doc`.

Reviewer: "This documentation will be gradually enhanced in future releases to cover all available methods and macros."
